algoliasearch/index_iterator.go

```diff
@@ -15,21 +13,17 @@ type IndexIterator struct {
 // newIndexIterator instantiates a IndexIterator on the `index` and according
 // to the given `params`. It is also trying to load the first page of results
 // and return an error if something goes wrong.
-func newIndexIterator(index *Index, params Map) (it IndexIterator, err error) {
-	it = IndexIterator{
+func newIndexIterator(index Index, params Map) (it IndexIterator, err error) {
+	it = &indexIterator{
 		index:  index,
 		params: duplicateMap(params),
 		pos:    0,
 	}
-	err = it.loadNextPage()
+	err = it.(*indexIterator).loadNextPage()
```

Comment:
I'm not entirely happy with this, but without avoiding the implicit return variable I don't see a better way.